lib/scaffold.dart

```diff
@@ -424,8 +420,8 @@ class BackdropScaffoldState extends State<BackdropScaffold>
 
   @override
   void dispose() {
+    _animationController?.dispose();
     super.dispose();
```

Comment:
The bug is not with `_shouldDisposeAnimationController`, it is calling of `_animationController?.dispose();` before `super.dispose` here, that has fixed the issue.

I suggest reverting back the `_shouldDisposeAnimationController` logic if the above is correct.

Reply:
Also, add a single line comment above `_shouldDisposeAnimationController` explaining its logic, something like `// not disposing of _animationController, if it was passed in widget.`

Reply:
Also the following:
```dart
if (widget.animationController == null && widget.controller == null) {
      _shouldDisposeAnimationController = true;
    }
```

could be simplified to:

```dart
// not disposing of `_animationController`, if it was passed in widget.
 _shouldDisposeAnimationController = (widget.animationController ?? widget.controller) == null;
```